.github/workflows/collection-continuous-integration.yml

```diff
@@ -0,0 +1,258 @@
+          - vm: ubuntu-16.04
+          - vm: macos-latest
+        python-version:
+          - 3.7
+          - 2.7
+
+    steps:
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Set up pip cache
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ github.ref }}-sanity-VMs
+        restore-keys: |
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-
+    - name: Install Ansible ${{ matrix.ansible-version }}
+      run: >-
+        python -m pip install --user ${{ matrix.ansible-version }} virtualenv
+    - name: Download migrated collection artifacts
+      uses: actions/download-artifact@v1
+      with:
+        name: >-
+          collection
+        path: .cache/collection-tarballs
+    - name: Install the collection tarball
+      run: >-
+        ~/.local/bin/ansible-galaxy collection install .cache/collection-tarballs/*.tar.gz
+    - name: Run collection sanity tests
+      run: >-
+        ~/.local/bin/ansible-test sanity --color --requirements --venv --python "${{ matrix.python-version }}" -vvv
+      working-directory: >-
+        /${{ runner.os == 'Linux' && 'home' || 'Users' }}/runner/.ansible/collections/ansible_collections/containers/podman
+
+  sanity-test-collection-via-containers:
+    name: Sanity in container via Python ${{ matrix.python-version }}
+    needs:
+    - build-collection-artifact
+    runs-on: ${{ matrix.runner-os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        runner-os:
+          - ubuntu-latest
+        runner-python-version:
+          - 3.7
+        ansible-version:
+          - ansible==2.9.5
+          - git+https://github.com/ansible/ansible.git@devel
+        python-version:
+          - 2.7
+          - 3.7
+
+    steps:
+    - name: Set up Python ${{ matrix.runner-python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.runner-python-version }}
+    - name: Set up pip cache
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ github.ref }}-sanity-containers
+        restore-keys: |
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-
+    - name: Install Ansible ${{ matrix.ansible-version }}
+      run: >-
+        python -m pip install --user ${{ matrix.ansible-version }}
+    - name: Download migrated collection artifacts
+      uses: actions/download-artifact@v1
+      with:
+        name: >-
+          collection
+        path: .cache/collection-tarballs
+    - name: Install the collection tarball
+      run: >-
+        ~/.local/bin/ansible-galaxy collection install .cache/collection-tarballs/*.tar.gz
+    - name: Run collection sanity tests
+      run: >-
+        ~/.local/bin/ansible-test sanity --color --requirements --docker --python "${{ matrix.python-version }}" -vvv
```

Comment:
I'm curious if it'd be faster w/o `--docker`. Also, `--python` arg probably isn't necessary.

Reply:
Oh, nevermind. I see what you're doing. OTOH you could install all the necessary Python interpreters on the VM directly... The benefit of using the pre-baked container is that it has more Python versions than GH Actions support. So you may want to reconsider.

Reply:
thanks, I'll try to get to it when optimizing jobs.